tests/vm_console_proxy_test.go

```diff
@@ -226,27 +246,87 @@ var _ = Describe("VM Console Proxy Operand", func() {
+			DeferCleanup(func() {
+				err := apiClient.Delete(ctx, serviceAccount)
+				if err != nil && !errors.IsNotFound(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
 
-		It("[test_id:9889] should be able to access /token endpoint", func() {
-			url, err := url.JoinPath(routeApiUrl, strategy.GetNamespace(), "non-existing-vm", "token")
+			role := &rbac.Role{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "proxy-test-role-",
+					Namespace:    strategy.GetNamespace(),
+				},
+				Rules: []rbac.PolicyRule{{
+					APIGroups: []string{"token.kubevirt.io"},
+					Resources: []string{"virtualmachines/vnc"},
+					Verbs:     []string{"get"},
+				}},
+			}
+			Expect(apiClient.Create(ctx, role)).To(Succeed())
+			DeferCleanup(func() {
+				err := apiClient.Delete(ctx, role)
+				if err != nil && !errors.IsNotFound(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
+
+			roleBinding := &rbac.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "proxy-role-binding-",
+					Namespace:    strategy.GetNamespace(),
+				},
+				Subjects: []rbac.Subject{{
+					Kind:      "ServiceAccount",
+					Name:      serviceAccount.Name,
+					Namespace: serviceAccount.Namespace,
+				}},
+				RoleRef: rbac.RoleRef{
+					APIGroup: rbac.GroupName,
+					Kind:     "Role",
+					Name:     role.Name,
+				},
+			}
+			Expect(apiClient.Create(ctx, roleBinding)).To(Succeed())
+			DeferCleanup(func() {
+				err := apiClient.Delete(ctx, roleBinding)
+				if err != nil && !errors.IsNotFound(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
+
+			tokenRequest, err := coreClient.CoreV1().ServiceAccounts(serviceAccount.Namespace).CreateToken(ctx, serviceAccount.Name, &authnv1.TokenRequest{}, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			// It may take a moment for the service to be reachable through route
+			saToken = tokenRequest.Status.Token
+		})
+
+		It("[test_id:TODO] should be able to access /vnc endpoint", func() {
+			vmNamespace := strategy.GetNamespace()
+			vmName := "non-existing-vm"
+
+			url := apiServerHostname + "/apis/token.kubevirt.io/v1alpha1/namespaces/" + vmNamespace + "/virtualmachines/" + vmName + "/vnc"
+
+			// It may take a moment for the service to be reachable
 			Eventually(func(g Gomega) {
-				response, err := httpClient.Get(url)
+				request, err := http.NewRequest("GET", url, nil)
 				g.Expect(err).ToNot(HaveOccurred())
-				defer func() { _ = response.Body.Close() }()
 
-				g.Expect(response.StatusCode).To(Equal(http.StatusUnauthorized))
+				request.Header.Set("Authorization", "Bearer "+saToken)
 
-				body, err := io.ReadAll(response.Body)
+				response, err := httpClient.Do(request)
 				g.Expect(err).ToNot(HaveOccurred())
+				defer func() { _ = response.Body.Close() }()
 
-				g.Expect(body).To(ContainSubstring("authenticating token cannot be empty"))
+				g.Expect(response.StatusCode).To(Equal(http.StatusUnauthorized))
```

Comment:
401 on successful access?

Reply:
Yes, the test only checks if the `vm-console-proxy` can be accessed. But maybe if the token is incorrect the API server also returns 401. I will check.

Reply:
I've changed the test.